src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/__tests__/ExposureDetectionViewModelTests.swift

```diff
@@ -637,9 +656,54 @@ class ExposureDetectionViewModelTests: XCTestCase {
 			isLoading: false
 		)
 	}
+	
+	func testIfRishChangeFromHighToLow_thenOTPisReseted() throws {
+		
+		let homeState = HomeState(
+			store: store,
+			riskProvider: MockRiskProvider(),
+			exposureManagerState: ExposureManagerState(authorized: true, enabled: true, status: .active),
+			enState: .enabled,
+			exposureSubmissionService: MockExposureSubmissionService(),
+			statisticsProvider: StatisticsProvider(
+				client: CachingHTTPClientMock(store: store),
+				store: store
+			)
+		)
+		homeState.updateDetectionMode(.automatic)
+		homeState.riskState = .detectionFailed
 
+		otpService.getOTP(ppacToken: PPACToken(apiToken: "test", deviceToken: "test")) { result in
+			switch result {
+			case .success:
+				XCTAssert(true)
+			case .failure(let error):
+				XCTFail(error.localizedDescription)
+			}
+		}
+		// we need the "let" or the viewModel instance will be deallocated when the expectations closure in line 701 returns
+		let viewModel = ExposureDetectionViewModel(
+			homeState: homeState,
+			appConfigurationProvider: CachedAppConfigurationMock(),
+			otpService: otpService,
+			onSurveyTap: { _ in },
+			onInactiveButtonTap: { _ in }
+		)
+		XCTAssertNotNil(viewModel, "View model should not be deallocated")
```

Comment:
Couldn't we just write… ? Or is the model emitting the `.riskStatusLowerd` notification?
```suggestion
		XCTAssertNotNil(ExposureDetectionViewModel(
		        homeState: homeState,
			appConfigurationProvider: CachedAppConfigurationMock(),
			otpService: otpService,
			onSurveyTap: { _ in },
			onInactiveButtonTap: { _ in }
		)
```

Reply:
The viewModel is system under test and resets the OTP (which is tested here)